ingest/workflow/snakemake_rules/transform.smk

```diff
@@ -0,0 +1,156 @@
+        [[ -f transform-authors ]]          || wget {params.transform_authors_url}
+        [[ -f apply-geolocation-rules ]]    || wget {params.apply_geolocation_rules_url}
+        [[ -f merge-user-metadata ]]        || wget {params.merge_user_metadata_url}
+        [[ -f ndjson-to-tsv-and-fasta ]]    || wget {params.ndjson_to_tsv_and_fasta_url}
+        chmod 755 *
+        cd ..
+
+        (cat {input.sequences_ndjson} \
+            | ./bin/transform-field-names \
+                --field-map {params.field_map} \
+            | ./bin/transform-string-fields --normalize \
+            | ./bin/transform-strain-names \
+                --strain-regex {params.strain_regex} \
+                --backup-fields {params.strain_backup_fields} \
+            | ./bin/transform-date-fields \
+                --date-fields {params.date_fields} \
+                --expected-date-formats {params.expected_date_formats} \
+            | ./bin/transform-genbank-location \
+            | ./bin/transform-string-fields \
+                --titlecase-fields {params.titlecase_fields} \
+                --articles {params.articles} \
+                --abbreviations {params.abbreviations} \
+            | ./bin/transform-authors \
+                --authors-field {params.authors_field} \
+                --default-value {params.authors_default_value} \
+                --abbr-authors-field {params.abbr_authors_field} \
+            | ./bin/apply-geolocation-rules \
+                --geolocation-rules {input.all_geolocation_rules} \
+            | ./bin/merge-user-metadata \
+                --annotations {params.annotations} \
+                --id-field {params.annotations_id} \
+            | ./bin/ndjson-to-tsv-and-fasta \
+                --metadata-columns {params.metadata_columns} \
+                --metadata {output.metadata} \
+                --fasta {output.sequences} \
+                --id-field {params.id_field} \
+                --sequence-field {params.sequence_field} ) 2>> {log}
+        """
+
+rule post_process_metadata:
+    input:
+        metadata="data/raw_metadata_{serotype}.tsv",
+    output:
+        metadata="data/metadata_{serotype}.tsv",
+    params:
+        post_process_metadata_url="https://raw.githubusercontent.com/nextstrain/zika/ingest/ingest/bin/post_process_metadata.py",
+
+    shell:
+        """
+        if [[ ! -d bin ]]; then
+          mkdir bin
+        fi
+        cd bin
+        [[ -f post_process_metadata.py ]] || wget {params.post_process_metadata_url}
+        chmod 755 *
+        cd ..
+
+        ./bin/post_process_metadata.py --metadata {input.metadata} --outfile {output.metadata}
+        """
+
+rule compress:
+    input:
+        sequences="data/sequences_{serotype}.fasta",
+        metadata="data/metadata_{serotype}.tsv",
+    output:
+        sequences="data/sequences_{serotype}.fasta.zst",
+        metadata="data/metadata_{serotype}.tsv.zst",
```

Comment:
Agree on serotype although this pattern also allows for multiple pathogen IDs to be fetched at once, or if only a subset of serotypes (with their own more specific NCBI Taxon ID) are of importance to the workflow. We can add it back in later if it becomes an issue but this may too overengineered for ebola.

Done with commit: fc124676868db197d697a9c4918ab0baaeb47a23

Reply:
Main ebola workflow now accepts and decompresses zst with commit: https://github.com/nextstrain/ebola/commit/92c96d630e7df5629d6340dfa5a8da09e2e12ae3